File: iosApp/iosApp/services/FirebaseAnalytics.swift
import Foundation
import shared
import FirebaseAnalytics

class FirebaseAnalytics: IFirebaseAnalytics {

    func logEvent(name: String) {
        Analytics.logEvent(name, parameters: nil)
    }

    func logEvent(name: String, params: [String : Any]) {
       Analytics.logEvent(name, parameters: params)
    }
    
    func resetAnalyticsData() {
        Analytics.resetAnalyticsData()
    }
    
    func setAnalyticsCollectionEnabled(enabled: Bool) {
        Analytics.setAnalyticsCollectionEnabled(enabled)
    }
    
    func setSessionTimeoutDuration(milliseconds: Int64) {
        Analytics.setSessionTimeoutInterval(TimeInterval(milliseconds/1000))
    }
    
    func setUserId(id: String?) {
        Analytics.setUserID(id)
    }
    
    func setUserProperty(name: String, value: String) {
        Analytics.setUserProperty(value, forName: name)
    }
}

File: androidApp/src/main/java/dev/haroldjose/familysharedlist/android/services/firebase/FirebaseAnalytics.kt
package dev.haroldjose.familysharedlist.android.services.firebase

import android.os.Bundle
import androidx.core.os.bundleOf
import com.google.firebase.Firebase
import com.google.firebase.analytics.analytics
import dev.haroldjose.familysharedlist.services.firebase.IFirebaseAnalytics

class FirebaseAnalytics: IFirebaseAnalytics {

    private var analytics = Firebase.analytics
    override fun logEvent(name: String) {
        analytics.logEvent(name, null)
    }

    override fun logEvent(name: String, params: Map<String, Any>) {
        val bundle = params.toBundle()
        analytics.logEvent(name, bundle)
    }

    override fun setAnalyticsCollectionEnabled(enabled: Boolean) {
        analytics.setAnalyticsCollectionEnabled(enabled)
    }

    override fun setUserId(id: String?) {
        analytics.setUserId(id)
    }

    override fun setSessionTimeoutDuration(milliseconds: Long) {
        analytics.setSessionTimeoutDuration(milliseconds)
    }

    override fun resetAnalyticsData() {
        analytics.resetAnalyticsData()
    }

    override fun setUserProperty(name: String, value: String) {
        analytics.setUserProperty(name, value)
    }
}

private fun Map<String, Any?>.toBundle(): Bundle = bundleOf(*this.toList().toTypedArray())